scanner/config/config.go

```diff
@@ -1,9 +1,143 @@
 package config
 
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	// DefaultConfiguration provides the default values for the scanner configuration.
+	DefaultConfiguration = Config{
+		HTTPListenAddr: ":9443",
+		GRPCListenAddr: ":8443",
+		Indexer: IndexerConfig{
+			Enable:       true,
+			DBConnString: "postgresql:///postgres?host=/var/run/postgresql",
+		},
+		Matcher: MatcherConfig{
+			Enable:       true,
+			DBConnString: "postgresql:///postgres?host=/var/run/postgresql",
+		},
+		// Default is empty.
+		MTLS: MTLSConfig{
+			CertsDir: "",
+		},
+		LogLevel: LogLevel(zerolog.InfoLevel),
+	}
+)
+
 // Config represents the Scanner configuration parameters.
 type Config struct {
-	// Mode specifies the mode in which Scanner will run.
-	//
-	// The options are: Indexer, Matcher, Combo.
-	Mode string `json:"mode"`
+	Indexer        IndexerConfig `yaml:"indexer"`
+	Matcher        MatcherConfig `yaml:"matcher"`
+	HTTPListenAddr string        `yaml:"http_listen_addr"`
+	GRPCListenAddr string        `yaml:"grpc_listen_addr"`
+	MTLS           MTLSConfig    `yaml:"mtls"`
+	LogLevel       LogLevel      `yaml:"log_level"`
+}
+
+func (c *Config) validate() error {
+	err := c.MTLS.validate()
+	if err != nil {
+		return fmt.Errorf("mtls: %w", err)
+	}
+	return nil
+}
+
+// IndexerConfig provides Scanner Indexer configuration.
+type IndexerConfig struct {
+	// Database provides indexer's database configuration.
+	DBConnString string `yaml:"db_conn_string"`
```

Comment:
I think Central hardcodes this (unless it is connecting to someone's BYODB). I think it may be worth asking why they chose to do that to see if we would want to consider that as well.

We could also opt to continue as-is and make it configurable, but then we could consider just using a hardcoded default (and then updating the default above to one which is more complete (has sslmode, password stuff, etc))

Reply:
> I think it may be worth asking why they chose to do that to see if we would want to consider that as well.

Hard coding is a subset of making it configurable with a default.

> We could also opt to continue as-is and make it configurable, but then we could consider just using a hardcoded default

Isn't t that what the patch is doing already?

> and then updating the default above to one which is more complete (has sslmode, password stuff, etc

The defaults were defined to run Scanner V4 in development/test mode, because that's where defaults are likely to make sense. In deployment, values should **always** be provided by the specs.